### PR TITLE
Move hierarchical facet options to the base Options class.

### PR DESCRIPTION
**Pull request by EreMaijala**
From #2372.

### CODE REVIEW

File: module/VuFind/src/VuFind/AjaxHandler/GetSideFacets.php [hunk 1]
```diff
@@ -241,11 +241,9 @@ protected function formatFacets(
         $hierarchicalFacets = [];
         $hierarchicalFacetSortOptions = [];
         $options = $results->getOptions();
-        if (is_callable([$options, 'getHierarchicalFacets'])) {
-            $hierarchicalFacets = $options->getHierarchicalFacets();
-            $hierarchicalFacetSortOptions
-                = $recommend->getHierarchicalFacetSortOptions();
-        }
+        $hierarchicalFacets = $options->getHierarchicalFacets();
```
Comment (demiankatz) [inline module/VuFind/src/VuFind/AjaxHandler/GetSideFacets.php:242]: You can now eliminate the empty array initializations of these variables above.

**Review by demiankatz — CHANGES_REQUESTED** [review]
One small suggestion...

**Comment by EreMaijala** [comment]
I decided to revert the changes to the constructors. Since almost everything else is also done only in the classes that actually use the functionality, it seems more clear to keep hierarchical facets that way as well (or alternatively move a lot of other settings init also to the base class' constructor).

File: module/VuFind/src/VuFind/AjaxHandler/GetSideFacets.php (hunk 1 above)
Comment (EreMaijala) [reply module/VuFind/src/VuFind/AjaxHandler/GetSideFacets.php:242]: Oops, done.

**Review by demiankatz — APPROVED** [review]

**Pull request merged**